src/Parameters/DocumentParametersTrait.php

```diff
@@ -0,0 +1,193 @@
+            $moduleNode = $xml->addChild('module');
+            $moduleNode->addAttribute('name', 'presentation');
+
+            foreach ($this->documents as $document) {
+                $documentNode = $moduleNode->addChild('document');
+
+                if (null === $documentNode) {
+                    throw new \Exception('XML could not be generated');
+                }
+
+                if ($document->getValidation()) {
+                    if (!$document->isValid()) {
+                        throw new \Exception("Document `{$document->getName()}` is not failed.");
+                    }
+                }
+
+                switch (get_class($document)) {
+                    case DocumentUrl::class:
+                        $documentNode->addAttribute('url', $document->getUrl());
+                        $documentNode->addAttribute('filename', $document->getName());
+
+                        break;
+
+                    case DocumentFile::class:
+                        $documentNode->addAttribute('name', $document->getName());
+                        $documentNode[0] = base64_encode($document->getFileContent());  // @phpstan-ignore-line
+
+                        break;
+
+                    default:
+                        throw new \Exception('The class `' . get_class($document) . '` is not a valid document.');
+                }
+
+                if (null !== $document->isCurrent()) {
+                    $value = $document->isCurrent() ? 'true' : 'false';
+                    $documentNode->addAttribute(DocumentOption::CURRENT->value, $value);
+                }
+
+                if (null !== $document->isDownloadable()) {
+                    $value = $document->isDownloadable() ? 'true' : 'false';
+                    $documentNode->addAttribute(DocumentOption::DOWNLOADABLE->value, $value);
+                }
+
+                if (null !== $document->isRemovable()) {
+                    $value = $document->isRemovable() ? 'true' : 'false';
+                    $documentNode->addAttribute(DocumentOption::REMOVABLE->value, $value);
+                }
+
+                foreach ($document->getProperties() as $propertyKey => $propertyValue) {
+                    $documentNode->addAttribute($propertyKey, $propertyValue);
+                }
+            }
+
+            $result = $xml->asXML();
+        }
+
+        if (!is_string($result)) {
+            throw new \RuntimeException('String expected, but ' . gettype($result) . ' received.');
+        }
+
+        return $result;
+    }
+
+    /**
+     * @throws \Exception
+     *
+     * @deprecated This function has been replaced by `addDocument`
+     */
+    public function addPresentation(string $nameOrUrl, ?string $content = null, ?string $filename = null, DocumentOptions $documentOptions = null): self
+    {
+        if (0 === mb_strpos($nameOrUrl, 'http')) {
+            $filename = $filename ?: 'unnamed file';
```

Comment:
We should not force the file naming if the API allows empty names. If there is any error the API will handle it for us.

Reply:
changed.